src/connectors/configure/__tests__/connectConfigure-test.ts

```diff
@@ -196,6 +225,52 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/configure/j
     );
   });
 
+  it('should not apply invalid userToken on refine()', () => {
+    const renderFn = jest.fn();
+    const createConfigure = connectConfigure(renderFn, jest.fn());
+    const configure = createConfigure({
+      searchParameters: {},
+    });
+    configure.init!(createInitOptions({ helper }));
+    const getUserToken = () =>
+      configure.getWidgetRenderState(createRenderOptions({})).widgetParams
+        .searchParameters.userToken;
+
+    const { refine } = configure.getWidgetRenderState(createRenderOptions({}));
+
+    // @ts-ignore-next-line
+    refine({ userToken: null });
+    expect(getUserToken()).toBeUndefined();
+
+    refine({ userToken: '' });
+    expect(getUserToken()).toBeUndefined();
+  });
+
+  it('should not override userToken with an invalid value on refine()', () => {
+    const renderFn = jest.fn();
+    const createConfigure = connectConfigure(renderFn, jest.fn());
+    const configure = createConfigure({
+      searchParameters: {},
+    });
+    configure.init!(createInitOptions({ helper }));
+    const getUserToken = () =>
+      configure.getWidgetRenderState(createRenderOptions({})).widgetParams
+        .searchParameters.userToken;
+
+    const { refine } = configure.getWidgetRenderState(createRenderOptions({}));
+
+    // set a valid userToken
+    refine({ userToken: 'abc' });
+    expect(getUserToken()).toEqual('abc');
+
+    // @ts-ignore-next-line
+    refine({ userToken: null });
+    expect(getUserToken()).toEqual('abc');
+
+    refine({ userToken: '' });
+    expect(getUserToken()).toEqual('abc');
```

Comment:
That behavior is very disturbing, as a user I would rather prefer to have an error when using an invalid userToken than a silent error with no explanation why my token was not updated.